netket/sampler/metropolis.py

```diff
@@ -176,6 +241,7 @@ def __pre_init__(self, hilbert: AbstractHilbert, rule: MetropolisRule, **kwargs)
                 `n_chains/mpi.n_nodes` chains. In general, we recommend specifying `n_chains_per_rank`
                 as it is more portable.
             n_chains_per_rank: Number of independent chains on every MPI rank (default = 16).
+            n_chains_per_device: Number of independent chains on every jax device (default = 16).
```

Comment:
I decided that for now the only simple solution is to separate the idea of devices and ranks. If we don't use MPI I need to have exactly 1 rank, and n_chains_per_rank needs to be the total number of chains (otherwise we have to change it everywhere, which I wouldn't mind we do later on)

If you want I can change the argument of this the function, but we have to live with the inconsistency that in the object itself the n_chains_per_rank property will then be n_chains_per_rank*n_devices, different from what you set

Reply:
but then you should be careful with defaults, does it make sense to have both default to 16?